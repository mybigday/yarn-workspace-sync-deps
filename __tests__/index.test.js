const fs = require('fs')
const path = require('path')
const shell = require('shelljs')

test('yarn-workspace-sync-deps', () => {
  const matchSnaps = () => {
    expect(
      fs.readFileSync(
        path.join(__dirname, 'fixture/test/test-package1/package.json'),
        'utf-8',
      ),
    ).toMatchSnapshot()
    expect(
      fs.readFileSync(
        path.join(__dirname, 'fixture/test2/test2-package1/package.json'),
        'utf-8',
      ),
    ).toMatchSnapshot()
    expect(
      fs.readFileSync(
        path.join(__dirname, 'fixture/test3/test3-package1/package.json'),
        'utf-8',
      ),
    ).toMatchSnapshot()
  }

  shell.cd(path.join(__dirname, 'fixture/'))
  matchSnaps()
  shell.exec('yarn')
  matchSnaps()
  shell.exec('git checkout .')
})
