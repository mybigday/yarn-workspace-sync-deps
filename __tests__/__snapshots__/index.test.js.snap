// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yarn-workspace-sync-deps 1`] = `
"{
  \\"name\\": \\"test-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"redux\\": \\"2\\"
  }
}
"
`;

exports[`yarn-workspace-sync-deps 2`] = `
"{
  \\"name\\": \\"test2-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"lodash\\": \\"3\\"
  }
}
"
`;

exports[`yarn-workspace-sync-deps 3`] = `
"{
  \\"name\\": \\"test3-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"lodash\\": \\"3\\"
  }
}
"
`;

exports[`yarn-workspace-sync-deps 4`] = `
"{
  \\"name\\": \\"test-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"redux\\": \\"^3.7.2\\"
  }
}
"
`;

exports[`yarn-workspace-sync-deps 5`] = `
"{
  \\"name\\": \\"test2-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"lodash\\": \\"^4.17.4\\"
  }
}
"
`;

exports[`yarn-workspace-sync-deps 6`] = `
"{
  \\"name\\": \\"test3-package1\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"lodash\\": \\"^4.17.4\\"
  }
}
"
`;
